sftp.go

```diff
@@ -227,3 +231,40 @@ func (s *StatusError) Error() string {
 func (s *StatusError) FxCode() fxerr {
 	return fxerr(s.Code)
 }
+
+// useful for tests
+func getEnabledExtensionByName(extensionName string) (sshExtensionPair, error) {
```

Comment:
This function seems to be _only_ used for testing. If it is going to be limited to tests, then it should be in the appropriate `*_test.go` file, rather than in non-test code.

I don’t think we should change the way the `assert.Equal` compares against the `sftpExtensions` as an array from how you had it before.

Especially, because otherwise the expectation would be that this function itself should be tested.

Reply:
> This function seems to be _only_ used for testing. If it is going to be limited to tests, then it should be in the appropriate `*_test.go` file, rather than in non-test code.

ok I can move it to tests or replace it with a function that allows to get the enabled extensions as []string. This method could be useful to to library users too and maybe we can make it public.

> 
> I don’t think we should change the way the `assert.Equal` compares against the `sftpExtensions` as an array from how you had it before.

the input is now `[]string` while the extenstions are `[]sshExtensionPair` we need the method explained above to get the enabled extensions as `[]string` to compare as before.

Please let me know what you prefer, thanks!

> 
> Especially, because otherwise the expectation would be that this function itself should be tested.

Reply:
Internal to the library, they should be using the `sftpExtensions` array directly.

The tests can test the value of `sftpExtensions` directly against an `expectedExtensions := []sshExtensionPair{…}` complex literal.